### DISCUSSION TITLE
mypy shows error for `session.exec(update)` and comparisons in `where` clause

### DISCUSSION THREAD

**Discussion by jpenna** [fastapi/sqlmodel#986] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import update
from pydantic import EmailStr
from sqlmodel import AutoString, Field, Relationship, SQLModel
from sqlmodel import Session

class User(SQLModel, table=True):
    email: EmailStr = Field(unique=True, index=True, sa_type=AutoString)
    full_name: str | None = None

def update_user(
    *,
    user: User,
    session: Session,
):
    session.exec(
        update(User)
        .where(
            User.email == "xxx",
            User.full_name == "yyy",
        )
        .values(
            email="nnn"
        )
    )
```

### Description

1. Use mypy for type checking
2. Copy the provided sample script
3. See the error messages when running mypy or using the mypy IDE plugin

### Operating System

Linux

### SQLModel Version

0.0.19

### Python Version

3.10.14

### Additional Context

What is the problem?

![Screenshot from 2024-06-15 13-22-10](https://github.com/tiangolo/sqlmodel/assets/16005946/18dcb085-db25-4975-9fc0-a17246d919ba)

**Comment by YuriiMotov** [comment]
For `where` clause you should use `col()`:
```py
from sqlmodel import col

...
select(User).where(col(User.email) == "xxx")
```

Regarding `session.exec(update(...)), there is open issue to track this: https://github.com/fastapi/sqlmodel/issues/909
